### PR TITLE
[SERV-323] Implement access token service for browser-based clients

### PR DESCRIPTION
**Pull request by markmatney**
See https://iiif.io/api/auth/1.0/#interaction-for-browser-based-client-applications for the feature specification.

### CODE REVIEW

File: src/main/java/edu/ucla/library/iiif/auth/TemplateKeys.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * The access token object key.
+     */
+    public static final String ACCESS_TOKEN_OBJECT = "access-token-object";
+
+    /**
+     * The campus network key.
+     */
+    public static final String CAMPUS_NETWORK = CookieJsonKeys.CAMPUS_NETWORK;
```
Comment (DRickard) [inline src/main/java/edu/ucla/library/iiif/auth/TemplateKeys.java:17]: Why have a new constant that has same value as CookieJsonKeys.CAMPUS_NETWORK? Why not just use CookieJsonKeys.CAMPUS_NETWORK?

Comment (markmatney) [reply src/main/java/edu/ucla/library/iiif/auth/TemplateKeys.java:17]: The idea is to have separate namespaces for each context that a "key" is used in:
- keys that are used to construct JSON response bodies;
- those used to construct cookies (which are encrypted JSON structures); and
- those used to pass values to Handlebars templates.

The intent is to provide an answer to the question: "which keys are valid for use in rendering Handlebars templates?" (or in constructing cookies, or in constructing JSON-valued HTTP responses).

In this case, the `cookie.hbs` template is used to render an HTML page that provides information to the user about the cookie they've just received. This statement you highlighted is an attempt to communicate that we intend to pass to the Handlebars template the same `CAMPUS_NETWORK` value that was used to construct the cookie. Similarly,

```java
public static final String ORIGIN = Param.ORIGIN;
```

is an attempt to communicate that we intend to pass the Handlebars template the same `ORIGIN` that was provided as a query parameter in the request URL.

It made sense to me and this is my reasoning, but I am open to doing this differently if it's convoluted or confusing. Maybe these classes would make more sense as enums instead?

Comment (markmatney) [reply src/main/java/edu/ucla/library/iiif/auth/TemplateKeys.java:17]: Although, maybe it _is_ a problem that some of these constants reference other constants? E.g., maybe we should be able to change the value of the `TemplateKeys.CAMPUS_NETWORK` without changing the corresponding cookie key...

Comment (markmatney) [reply src/main/java/edu/ucla/library/iiif/auth/TemplateKeys.java]: Okay, the latest commit decouples all the IDs that aren't required to have the same value (`ResponseJsonKeys.MESSAGE_ID` is the exception).

**Comment by ksclarke** [comment]
Seems we don't have Snyk hooked up to CI yet, but running locally I see some issues (some of which can be addressed):

[INFO] Issues to fix by upgrading:
[INFO] 
[INFO]   Upgrade ch.qos.logback:logback-classic@1.2.5 to ch.qos.logback:logback-classic@1.2.7 to fix
[INFO]   ✗ Insufficient Hostname Verification [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-CHQOSLOGBACK-1726923] in ch.qos.logback:logback-core@1.2.5
[INFO]     introduced by ch.qos.logback:logback-classic@1.2.5 > ch.qos.logback:logback-core@1.2.5
[INFO] 
[INFO] 
[INFO] Issues with no direct upgrade or patch:
[INFO]   ✗ Denial of Service (DoS) [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-2326698] in com.fasterxml.jackson.core:jackson-databind@2.11.4
[INFO]     introduced by io.vertx:vertx-web-openapi@4.1.5 > com.fasterxml.jackson.core:jackson-databind@2.11.4
[INFO]   This issue was fixed in versions: 2.13.1, 2.12.6
[INFO]   ✗ Improper Certificate Validation [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-1042268] in io.netty:netty-handler@4.1.68.Final
[INFO]     introduced by io.vertx:vertx-core@4.1.5 > io.netty:netty-handler@4.1.68.Final
[INFO]   No upgrade or patch available
[INFO]   ✗ HTTP Request Smuggling [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-IONETTY-2314893] in io.netty:netty-codec-http@4.1.68.Final
[INFO]     introduced by io.vertx:vertx-core@4.1.5 > io.netty:netty-codec-http@4.1.68.Final
[INFO]   This issue was fixed in versions: 4.1.71.Final

**Review by ksclarke — APPROVED** [review]
I'm a little concerned about our variation in naming conventions between path, query, cookie, etc., since our hauth.yml has some variation between kebab and camel case, but I've created a ticket to document a comprehensive story about when we use which, and where: https://jira.library.ucla.edu/browse/SERV-328 -- I just want to make sure we're consistent across projects.

**Pull request merged**